File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/actions/IntakePitchDown.java
package org.firstinspires.ftc.teamcode.robot.actions;

import androidx.annotation.NonNull;

import com.acmerobotics.dashboard.telemetry.TelemetryPacket;
import com.acmerobotics.roadrunner.Action;

import org.firstinspires.ftc.teamcode.robot.Intake;

public class IntakePitchDown implements Action {
    private boolean init = false;
    private Intake i;

    @Override
    public boolean run(@NonNull TelemetryPacket packet) {
        if(!init) {
            i.pitchDown();
            init = true;
        }

        return false;
    }

    public IntakePitchDown(Intake li) {
        i = li;
    }

    public Action intakePitchDown(Intake li) {
        return new org.firstinspires.ftc.teamcode.dairy.actions.IntakePitchDown(li);
    }
}

File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/dairy/actions/IntakePitchDown.kt
package org.firstinspires.ftc.teamcode.dairy.actions

import com.acmerobotics.dashboard.telemetry.TelemetryPacket
import com.acmerobotics.roadrunner.Action
import org.firstinspires.ftc.teamcode.dairy.subsystems.IntakeClaw
import org.firstinspires.ftc.teamcode.dairy.util.Waiter

class IntakePitchDown() : Action {
    private var init = false
    private lateinit var waiter:Waiter

    override fun run(p: TelemetryPacket): Boolean {
        if (!init) {
            waiter = Waiter()
            waiter.start(400)
            IntakeClaw.pitch_pos = IntakeClaw.pitch_down
            init = true
        }

        return waiter.isDone
    }

    companion object {
    fun intakePitchDown(): Action {
        return IntakePitchDown()
    }}
}